### DISCUSSION TITLE
trying to import jpype in AWS lambda function

### DISCUSSION THREAD

**Discussion by GuptaNaman1998** [jpype-project/jpype#1103] [Q&A] [answered]
[ERROR] Runtime.ImportModuleError: Unable to import module 'lambda_function': No module named '_jpype'
Traceback (most recent call last):

I checked the jpype folder and saw that _init_.py file has a statement "import _jpype"
but in the jpype there was no file named _jpype.py

Version is 1.4.1 in the init file.

**Comment by Thrameos** [comment]
All of the wheels ship with ``_jpype.{arch}.so``.   It is the shared library that is used for the back end.   It seems like your site has a broken install which is missing the backend.

JPype install requires ``_jpype.so``, ``org.jpype.jar`` and ``jpype\`` in the site install directory.

**Reply by GuptaNaman1998** [reply]
Thank you for helping. I was trying to startJVM using jpype on AWS Lambda but the code is throwing the below error :
"error": "No JVM shared library file (libjvm.so) found. Try setting up the JAVA_HOME environment variable properly."
I was able to resolve Unable to import module 'lambda_function': No module named '_jpype' error by reinstalling jpype.

**Reply by vigneshr3212** [reply]
Hi @GuptaNaman1998 ... Im also facing the same issue..**Unable to import module 'lambda_function': No module named '_jpype'** can u please help me out. My deployment package structure is 
![Uploading Screenshot 2024-01-31 at 9.32.20 PM.png…]()

**Reply by GuptaNaman1998** [reply]
Hi @vigneshr3212 ,

The resolution to my issue was via 2 routes. ( please bear in mind that my objective of using jpype was to connect to databricks)

1. AWS previously used to preinstall the JVM package on the underlying machines running lambdas, but with the newer releases in order to reduce the machine storage on which the lambda runs, they removed the JVM altogether. (python 3.7 and lower should have this JVM installed)
2. As per the above point, AWS Lambda functions run on amazon-linux images and it may sound silly but compiling or packaging the jpype in linux machine can solve this.

pro tip : use docker to get these packages.

**Reply by vigneshr3212** [reply]
Hi @GuptaNaman1998, Thanks for your inputs, the error with jpype has resolved, but now im facing the erros JVM module no found, eventhough i package the JVM in linux machine 
{
  "errorMessage": "[Errno 0] JVM DLL not found: /var/task",
  "errorType": "OSError",
  "requestId": "275852d2-72f6-49fd-b900-d3d2aa7a94de",
  "stackTrace": [
    "  File \"/var/task/lambda_function.py\", line 6, in lambda_handler\n    jpype.startJVM(convertStrings=False, jvmpath='/var/task')\n",
    "  File \"/var/task/jpype/_core.py\", line 247, in startJVM\n    _jpype.startup(jvmpath, jvmargs + extra_jvm_args,\n"
  ]
}

**Reply by GuptaNaman1998** [reply, marked as answer]
Hi @vigneshr3212 
What is the lambda version you are using?
Error : jpype._jvmfinder.JVMNotFoundException: No JVM shared library file (jvm.dll) found. Try setting up the JAVA_HOME environment variable properly.
Resolution :
- It was communicated to me that this was an issue with AWS
- In an attempt to save space, the lambda python 3.8+ runtimes did not have JVM installed in them

So in order to make it work you'll have to write a code that will install the jvm and then use it or maybe use an instance with lambda python 3.7.
